### PR TITLE
DEV: add plugin hooks for silence message parameters

### PR DESCRIPTION
**Pull request by featheredtoast**
Allows plugins to add, and update extra silence message params for custom
i18n vars

### CODE REVIEW

File: app/services/user_silencer.rb [hunk 1]
```diff
@@ -57,8 +57,11 @@ def silence
           user_history: @user_history,
           post_id: @opts[:post_id],
           silenced_till: @user.silenced_till,
-          silenced_at: DateTime.now
+          silenced_at: DateTime.now,
+          silence_message_params: silence_message_params
         )
+
+        SystemMessage.create(@user, message_type, **silence_message_params)
```
Comment (ZogStriP) [inline app/services/user_silencer.rb:64]: Why do you need the splat operator here?

Comment (featheredtoast) [reply app/services/user_silencer.rb:64]: Ruby 2.7.1 deferred this (no longer warns), but there was previous a bunch of warnings in 2.7.0 about how passing an options hash as a variable (and having it auto-convert to keyword args) was deprecated: https://blog.bigbinary.com/2020/04/14/ruby-2-7-deprecates-conversion-of-keyword-arguments.html

The splat operator makes that conversion explicit.

Comment (featheredtoast) [reply app/services/user_silencer.rb]: Wait, you're right, there's no reason for it - I was misreading the function definition, and thought the args hash argument was a keyword hash.

File: app/services/user_silencer.rb [hunk 2]
```diff
@@ -57,8 +57,11 @@ def silence
           user_history: @user_history,
           post_id: @opts[:post_id],
           silenced_till: @user.silenced_till,
-          silenced_at: DateTime.now
+          silenced_at: DateTime.now,
+          silence_message_params: silence_message_params
```
Comment (tgxworld) [inline app/services/user_silencer.rb:61]: Let's add a test for this param as well to demonstrate the intended usage :)

Comment (featheredtoast) [reply app/services/user_silencer.rb:61]: Oh right sorry aobut that - just added!

**Pull request merged**